fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/app/api/sensor/BasicSensorModule.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.fed.application.app.api.sensor;
+
+import org.eclipse.mosaic.lib.enums.SensorType;
+
+/**
+ * A basic sensor module which provides single integer values for a given {@link SensorType}.
+ */
+public interface BasicSensorModule {
+
+    /**
+     * Enables this basic sensor module.
+     */
+    void enable();
+
+    /**
+     * @return {@code true}, if this module has been enabled.
+     */
+    boolean isEnabled();
+
+    /**
+     * Disables this basic sensor module. {@link #getStrengthOf(SensorType)} will always return 0.
+     */
+    void disable();
+
+    /**
+     * Returns the state (as strength) of the supplied environment {@link SensorType}.
+     *
+     * @param sensorType The {@link SensorType} type to use.
+     * @return Strength of the measured environment sensor data.
+     */
+    int getStrengthOf(SensorType sensorType);
```

Comment:
Apart from the interface design, I believe that the sensor module for environmental situations mainly lives from the `SensorType`.

We might spend additional efforts in defining the SensorTypes better, e.g. according to current state of DENM content definitions, whereas these ASN.1 definifions in the ROS documentation could help:
https://docs.ros.org/en/jazzy/p/etsi_its_vam_ts_msgs/interfaces/message_definitions.html

Additionally, we might get over this `getStrengthOf (sensorType)` modelling by renaming this to more generic `getValue (sensorType)` or `getValue (sensorType)`

Reply:
Needs definitely a rework in a follow up.